Jenkinsfile

```diff
@@ -65,18 +67,107 @@ pipeline {
+                }
+            }
+            steps {
+                script {
+                    dir('deploy/docker/notebook/stacks') {
+                        withEnv(["HOME=${env.WORKSPACE}"]) {
+                            sh 'mkdir -p manifests'
+
+                            stacks = [
+                                ['Python-datascience.yaml', 'Python-dataviz.yaml'],
+                                ['R.yaml'],
+                                ['octave.yaml'],
+                                ['java.yaml', 'scala.yaml'],
+                                ['cpp.yaml'],
+                                ['bash.yaml'],
+                                ['tensorflow.yaml', 'pytorch.yaml', 'fastai.yaml'],
+                                ['latex.yaml']
+                            ]
+
+                            // CPU-based images
+                            // Image without additional stacks
+                            sh 'railyard assemble -t Dockerfile.template -b base.yaml -p manifests'
+
+                            // Images with a single additional stack
+                            stacks.each {
+                                sh "railyard assemble -t Dockerfile.template -b base.yaml " + it.collect{"-a " + it}.join(" ") + " -p manifests"
+                            }
+
+                            // Images with combinations of 2 additional stacks
+                            [stacks, stacks].combinations().findAll{item -> item[0].join(" ") < item[1].join(" ")}.collect{it.flatten()}.each {
+                                sh "railyard assemble -t Dockerfile.template -b base.yaml " + it.collect{"-a " + it}.join(" ") + " -p manifests"
+                            }
+
+                            // GPU-based images
+                            // Image without additional stacks
+                            sh 'railyard assemble -t Dockerfile.template -b base_gpu.yaml -p manifests'
+
+                            // Images with a single additional stack
+                            stacks.each {
+                                sh "railyard assemble -t Dockerfile.template -b base_gpu.yaml " + it.collect{"-a " + it}.join(" ") + " -p manifests"
+                            }
+
+                            // Images with combinations of 2 additional stacks
+                            [stacks, stacks].combinations().findAll{item -> item[0].join(" ") < item[1].join(" ")}.collect{it.flatten()}.each {
+                                sh "railyard assemble -t Dockerfile.template -b base_gpu.yaml " + it.collect{"-a " + it}.join(" ") + " -p manifests"
+                            }
+                        }
+                    }
+                }
+            }
+        }
         stage('Build Jupyter Notebook Docker') {
             when {
                 environment name: 'SKIP_BUILD', value: 'false'
-                environment name: 'BUILD_NOTEBOOK', value: '0'
+                // environment name: 'BUILD_NOTEBOOK', value: '0'
```

Comment:
This can be removed. The BUILD_NOTEBOOK variable can also be removed in the above `environments` section.

Reply:
Do you suggest we use different image tags for CPU/GPU? I.e.
```
labshare/polyglot-notebook/cpu-<hash>
labshare/polyglot-notebook/gpu-<cuda_version>-<hash>
```

I don't see much utility in that, as hash already encompasses all of that information (i.e. two similar images using cpu base vs gpu base will have a different hash)